packages/frontend/src/context/positions.tsx

```diff
@@ -97,45 +99,32 @@ const PositionsProvider: React.FC = ({ children }) => {
     return { shortRealizedPNL: pnlForOneSqth.multipliedBy(boughtSqueeth) }
   }, [boughtSqueeth.toString(), totalUSDFromBuy.toString(), soldSqueeth.toString(), totalUSDFromSell.toString()])
 
-  const mintedDebt = useMemo(() => {
-    // squeethAmount = user long balance if oSqueethBal > 0, but it could also be minted balance
-    return shortVaults[firstValidVault]?.shortAmount.gt(0) &&
-      oSqueethBal?.isGreaterThan(0) &&
-      positionType === PositionType.LONG
-      ? oSqueethBal.minus(squeethAmount)
-      : shortVaults[firstValidVault]?.shortAmount.gt(0) && oSqueethBal?.isGreaterThan(0)
-      ? oSqueethBal
+  //when the squeethAmount < 0 and the abs amount is greater than openShortSqueeth, that means there is manually sold short position
+  const mintedSoldShort = useMemo(() => {
+    return positionType === PositionType.SHORT && squeethAmount.abs().isGreaterThan(openShortSqueeth)
+      ? squeethAmount.abs().minus(openShortSqueeth)
       : new BigNumber(0)
-  }, [firstValidVault, oSqueethBal?.toString(), positionType, shortVaults?.length, squeethAmount.toString()])
-
-  const shortDebt = useMemo(() => {
-    return positionType === PositionType.SHORT ? squeethAmount : new BigNumber(0)
-  }, [positionType, squeethAmount.toString()])
-
-  const longSqthBal = useMemo(() => {
-    return mintedDebt.gt(0) ? oSqueethBal.minus(mintedDebt) : oSqueethBal
-  }, [oSqueethBal?.toString(), mintedDebt.toString()])
+  }, [positionType, squeethAmount?.toString(), openShortSqueeth.toString()])
 
   const lpDebt = useMemo(() => {
     return depositedSqueeth.minus(withdrawnSqueeth).isGreaterThan(0)
       ? depositedSqueeth.minus(withdrawnSqueeth)
       : new BigNumber(0)
   }, [depositedSqueeth.toString(), withdrawnSqueeth.toString()])
 
-  const { finalSqueeth, finalWeth } = useMemo(() => {
-    // dont include LPed & minted amount will be the correct short amount
-    const finalSqueeth = squeethAmount
-    const finalWeth = wethAmount.div(squeethAmount).multipliedBy(finalSqueeth)
-    return { finalSqueeth, finalWeth }
-  }, [squeethAmount.toString(), wethAmount.toString()])
+  //mintedSqueeth balance - mintedSold short position = existing mintedDebt in vault
+  //since LPed amount wont be deducted, will need to be deducted here
```

Comment:
looks like LP is not deducted in the below calc though? Is that intended? Maybe im just confused at the comment haha

Reply:
it is in line 118, which is mintedSqueeth.minus(mintedSoldShort).minus(depositedSqueeth).plus(withdrawnSqueeth) = minted squeeth, and I will update the comment to make it more clear lol

Reply:
Ok gotcha, LP is deducted as part of which var in that equation?

Reply:
With .minus(depositedSqueeth).plus(withdrawnSqueeth) in line 118, depositedSqueeth.plus(withdrawnSqueeth) would be LPed amount

Reply:
ok maybe add a comment documenting that? i def wouldn't have known that so prob helpful for others 

what does depositedSqueeth represent and what does withdrawnSqueeth represent? Var names are a bit confusing imo

Reply:
Just added!

depositedSqueeth is accumulated deposited Squeeth in LP position and withdrawn Squeeth is acc. withdrawn squeeth in LP position, and accumulated deposited Squeeth - acc. withdrawn squeeth in LP position = LP debt